site/content/user-guide/amqp-adapter.md

```diff
@@ -0,0 +1,175 @@
++++
+title = "AMQP Adapter"
+weight = 215
++++
+
+The AMQP protocol adapter allows clients (devices or gateway components) supporting the AMQP 1.0 protocol to publish telemetry messages and events to Eclipse Hono&trade;'s Telemetry and Event endpoints. The AMQP adapter supports uploading telemetry messages and events using the `AT_LEAST_ONCE` delivery semantics. It also supports uploading telemetry messages using the `AT_MOST_ONCE` (fire and forget) delivery semantics. With `AT_MOST_ONCE` delivery, the client device receives the *ACCEPTED* disposition frame from the adapter (even if the message cannot be delivered to the downstream consumer peer). This is because with `AT_MOST_ONCE`, the client has indicated that it is not interested to know what happens to the message once it has being sent. The `AT_MOST_ONCE` delivery semantics is **not supported** for events because with events, clients MUST be notified regardless of weather the message was delivered to the downstream consumer or not (in the case of a failure).
+
+On the other hand, with `AT_LEAST_ONCE` delivery, clients indicate that they want to get a response if the message was successfully delivered to the downstream peer and, in the case of failure, be informed that the message cannot be delivered. To this end, the AMQP adapter distinguishes between two types of failures:
+
+* A failure caused by a client-side error (e.g invalid message content type, missing device-id in address, adapter disabled for tenant, etc).
+* A failure caused by a server-side error (e.g sender queue full, no credit available yet to send the message, etc).
+
+For a client-side error, the adapter responds with the *REJECTED* disposition and provides a reason why the message was rejected together with the error condition value. If a bad request is the reason why the message cannot be processed, then the adapter returns a `hono:bad-request` as the error condition value. If the message cannot be processed because the adapter was disabled for the tenant, then the adapter returns the `amqp:unauthorized-access` error condition value. For all other client-side errors, the adapter returns the `amqp:precondition-failed` as error condition value. 
+
+In the case of a server-side error, the adapter responds with a *RELEASED* disposition, which indicates to the client that the message was OK but it cannot be delivered due to a failure beyond the control of the client. In this case, the client should attempt to redeliver the message again.
+
+<!--more-->
```

Comment:
this needs to go waaaaaayyy up. It is the separator between the first one or two sentences explaining what this page is about and the rest of the page ...

Reply:
ok

Reply:
please move to line 7